src/main/java/ladder/domain/Names.java

```diff
@@ -20,25 +21,49 @@ public Names(String names) {
             .collect(Collectors.toList());
     }
 
+    public Names(String names, int expectedCount) {
+        this(names);
+        validateNamesCount(expectedCount);
+    }
+
     private void validateNames(String names) {
-        if (isNull(names)) {
-            throw new IllegalArgumentException("이름이 null 입니다");
+        validateNull(names);
+        validateFormat(names);
+    }
+
+    private void validateNull(String names) {
+        if (names == null) {
+            throw new IllegalArgumentException("아무 값도 입력되지 않았습니다.");
         }
-        if (isFormat(names)) {
+    }
+
+    private void validateFormat(String names) {
+        if (!NAMES_FORMAT.matcher(names).matches()) {
             throw new IllegalArgumentException("이름이 형식과 맞지 않습니다");
         }
     }
 
-    private boolean isNull(String names) {
-        return names == null;
+    private void validateNamesCount(int expectedCount) {
+        if (this.names.size() != expectedCount) {
+            throw new IllegalArgumentException(
+                String.format("입력하는 이름의 개수는 %d와 같아야 합니다.(입력된 이름의 수 : %d)",
+                    expectedCount, this.names.size()));
+        }
+    }
+
+    public int findNamesCount() {
+        return names.size();
     }
 
-    private boolean isFormat(String names) {
-        return !NAMES_FORMAT.matcher(names).matches();
+    public Name findNameByIndex(int index) {
+        return names.get(index);
     }
 
-    public int getCount() {
-        return names.size();
+    public Name findName(String name) throws IllegalArgumentException {
+        return names.stream()
+            .filter((el) -> el.isEqual(name))
+            .findFirst().orElseThrow(
+                () -> new IllegalArgumentException("찾으려는 이름이 이름 목록에 존재하지 않습니다."));
     }
```

Comment:
https://github.com/woowacourse/java-ladder/pull/195#discussion_r1116528790 https://github.com/woowacourse/java-ladder/pull/195#discussion_r1118035577
해당 코멘트의 답변 남겨드립니다~

```suggestion
    public Name findName(String rawName) throws IllegalArgumentException {
        final var name = new Name(rawName);
        if (!names.contains(name)) {
            throw new IllegalArgumentException("찾으려는 이름이 이름 목록에 존재하지 않습니다.");
        }
        
        return name;
    }
```

해당 메서드의 기능을 풀어보면 위와 같을 것 같습니다! 
아마 개선하는 과정에서 애매하게 느껴진 이유가 해당 기능의 목적이 애매해서 그런 것 같은데요.

https://github.com/woowacourse/java-ladder/pull/195#discussion_r1118029544

해당 코멘트와 연결해서 `위 기능이 어떠한 목적인가?`로 보면 역할이 애매해서 명확하게 변경해보면 좋을 것 같습니다 :)

Reply:
Name에 대한 equals 메서드를 오버라이딩해서 contains로 처리할 수 있도록 구현했습니다!